### PR TITLE
[wasm] Decrease frequency of chrome version bumps

### PR DESCRIPTION
**Pull request by ilonatommy**
Partial fix for https://github.com/dotnet/runtime/issues/97364.

Added logic:
- Read current `ChromeVersions.props`
- Compare the latest versions with current
- If any platform, Linux or Windows has updated `ChromeBaseSnapshotUrl` or `V8Version`, then update the `*.props`
- Otherwise, log that we're skipping an update.

## Out of scope
Investigate why we can't find more snapshot urls - from this reason the original issue should stay opened.

### CODE REVIEW

File: src/tasks/WasmBuildTasks/GetChromeVersions.cs [hunk 1]
```diff
@@ -90,6 +97,31 @@ private async Task<bool> ExecuteInternalAsync()
         }
     }
 
+    private bool AreVersionsChanged(ChromeVersionSpec version, string baseUrl)
+    {
+        XmlDocument xmlDoc = new XmlDocument();
+        xmlDoc.Load(ChromeVersionsPath);
+
+        if (version.os == "Linux")
```
Comment (akoeplinger) [inline src/tasks/WasmBuildTasks/GetChromeVersions.cs:105]: nit: use case insensitive comparison

File: src/tasks/WasmBuildTasks/GetChromeVersions.cs [hunk 2]
```diff
@@ -90,6 +97,31 @@ private async Task<bool> ExecuteInternalAsync()
...
+        {
+            string linuxChromeBaseSnapshotUrl = GetNodeValue(xmlDoc, "linux_ChromeBaseSnapshotUrl");
+            string linuxV8Version = GetNodeValue(xmlDoc, "linux_V8Version");
+            return version.v8_version != linuxV8Version ||
+                baseUrl != linuxChromeBaseSnapshotUrl;
+        }
+        // version.os == "Windows"
```
Comment (akoeplinger) [inline src/tasks/WasmBuildTasks/GetChromeVersions.cs:112]: I'd prefer having an else if for the windows case and add an else+throw if os is not recognized

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
I noticed that https://omahaproxy.appspot.com which is used in s_allJsonUrl/s_historyUrl/etc is no longer available, maybe a reason for the issue mentioned in https://github.com/dotnet/runtime/issues/97364

File: eng/testing/bump-chrome-version.proj [hunk 3]
```diff
@@ -2,39 +2,46 @@
...
 
   <Target Name="UpdateChromeVersion">
     <GetChromeVersions
                 OSIdentifier="linux"
                 OSPrefix="Linux_x64"
                 Channel="$(ChromeChannel)"
                 MaxMajorVersionsToCheck="1"
-                IntermediateOutputPath="$(ArtifactsObjDir)">
+                IntermediateOutputPath="$(ArtifactsObjDir)"
+                ChromeVersionsPath="$(ChromeVersionsPath)">
```
Comment (radical) [inline eng/testing/bump-chrome-version.proj:16]: This kinda exposes the props file to the `*Get*ChromeVersions` task, which if we want to do then it should also do the writing, and be renamed to `UpdateChromeVersions`. And the task can emit a property for `VersionsChanged`.

A different approach would be to have the versions from the task in temporary properties. Then compare against the existing versions/urls from the props file, and then decide what to do. Essentially, moving that bit out to the targets itself.

**Comment by radical** [comment]
> I noticed that [omahaproxy.appspot.com](https://omahaproxy.appspot.com) which is used in s_allJsonUrl/s_historyUrl/etc is no longer available, maybe a reason for the issue mentioned in #97364

We moved to using chromiumdash to get the versions. We can drop the omahaproxy code in the task now.

File: eng/testing/bump-chrome-version.proj (hunk 3 above)
Comment (ilonatommy) [reply eng/testing/bump-chrome-version.proj:16]: Taking option 1, `UpdateChromeVersions` task that is doing everything and returning info for `Message` if there was a change of not.

**Review by akoeplinger — APPROVED** [review]

**Comment by ilonatommy** [comment]
Library tests were not touched in this PR and they look like a timeout. Merging without them

**Pull request merged**